spec/controllers/courses_controller_spec.rb

```diff
@@ -62,9 +77,16 @@
   describe "GET #edit" do
     it "assigns the requested course as @course" do
       course = Course.create! valid_attributes
+      FactoryGirl.create(:course_participant, user: user, course: course)
       get :edit, {:id => course.to_param}, valid_session
       expect(assigns(:course)).to eq(course)
     end
+    it "404s if it's not your course" do
```

Comment:
not a big fan of just getting ActiveRecord::RecordNotFound here... that eventually rises up to 404 (right?), but it might make more sense to `render nothing: true, status: 404`